x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java

```diff
@@ -0,0 +1,481 @@
+            subParser = new XContentSubParser(context.parser());
+            token = subParser.nextToken();
+            while (token != XContentParser.Token.END_OBJECT) {
+                // should be an object subfield with name a metric name
+                ensureExpectedToken(XContentParser.Token.FIELD_NAME, token, subParser::getTokenLocation);
+                String fieldName = subParser.currentName();
+                Metric metric = Metric.valueOf(fieldName);
+
+                if (metrics.value().contains(metric) == false) {
+                    throw new IllegalArgumentException("Aggregate metric [" + metric +
+                        "] does not exist in the mapping of field [" + fieldType.name() + "]");
+                }
+
+                token = subParser.nextToken();
+                // Make sure that the value is a number. Probably this will change when
+                // new aggregate metric types are added (histogram, cardinality etc)
+                ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token, subParser::getTokenLocation);
+                NumberFieldMapper delegateFieldMapper = metricFieldMappers.get(metric);
+                delegateFieldMapper.parse(context);
+
+                if (Metric.value_count == metric) {
+                    Number n = context.doc().getField(delegateFieldMapper.fieldType().name()).numericValue();
+                    if  (n.intValue() < 0) {
+                        throw new IllegalArgumentException("Aggregate metric [" + metric.name() +
+                            "] of field [" + fieldType.name() + "] cannot be a negative number");
+                    }
+                }
+
+                token = subParser.nextToken();
+            }
+
+            for (Metric m : metrics.value()) {
+                if (context.doc().getField(fieldType().name() + "._" + m.name()) == null) {
+                    throw new IllegalArgumentException("Aggregate metric field [" + fieldType.name() +
+                        "] must contain all metrics " + metrics.value().toString());
+                }
+            }
+        } catch (Exception e) {
+            if (ignoreMalformed.value() == true) {
+                if (subParser != null) {
+                    // close the subParser so we advance to the end of the object
+                    subParser.close();
+                }
+                context.addIgnoredField(fieldType().name());
+            } else {
+                // Rethrow exception as is. It is going to be caught and nested in a MapperParsingException
+                // by its FieldMapper.MappedFieldType#parse()
+                throw e;
+            }
+        }
+        context.path().remove();
+    }
+
+    @Override
+    protected void doMerge(Mapper mergeWith) {
+        super.doMerge(mergeWith);
+        AggregateDoubleMetricFieldMapper other = (AggregateDoubleMetricFieldMapper) mergeWith;
```

Comment:
Not sure but I would expect that merging two fields with different metrics should be an error.